### ISSUE TITLE
Random 404 responses with Traefik v.2.2.2

### ISSUE THREAD

**Issue by floatingpurr** [containous/traefik#7024]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

I'm running Traefik in Docker. I'm using the image v.2.2.2 (upgrading from v.2.2.1). Oddly enough, sometimes Traefik returns HTTP 404 responses. This happens randomly and at the same time on different Web browsers. For example, I open up `test.foo.org` with Google Chrome and I get a 404. Immediately after, I open the same url with Safari and it works. If I refresh the page with Google Chrome, It keeps not working.

Reverting to v.2.2.1 removes this issue

Coming back to v.2.2.2 reintroduces that problem

### What did you expect to see?
2XX responses for defined hosts.

### What did you see instead?
Sometimes Traefik returns a 404. E.g.,

Request
```
Request URL: https://test.foo.org/
Request Method: GET
Status Code: 404 
Remote Address: XXX.XXX.XXX.XXX:443
Referrer Policy: no-referrer-when-downgrade
```

Response
```
content-length: 19
content-type: text/plain; charset=utf-8
date: Fri, 10 Jul 2020 13:57:38 GMT
status: 404
x-content-type-options: nosniff
```

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      2.2.2
Codename:     chevrotin
Go version:   go1.14.4
Built:        2020-07-08T15:30:29Z
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

Docker installation on Ubuntu 20 LTS. Here is my `docker-compose`:

```yml
version: '3.7'

services:
  # a cool reverse-proxy / load balancer
  traefik:
    # The official v2 Traefik docker image
    image: traefik:v2.2
    container_name: traefik
    security_opt:
      - no-new-privileges:true
    restart: always
    env_file:
      # variables that will be passed to the container
      - .provider.env
    volumes:
      # So that Traefik can listen to the Docker events
      - /var/run/docker.sock:/var/run/docker.sock:ro
      # tls-certificates path (TLS)
      - ./letsencrypt:/letsencrypt
    ports:
      # http
      - ${HTTP_PORT}:80
      # https
      - ${HTTPS_PORT}:443
    command:
      ###########################################
      #   Static Configuration harnessing CLI   #
      ###########################################
      # Activate dashboard.
      - --api.dashboard=true

      # Log level set to traefik logs. (Default: ERROR)
      - --log.level=${DEBUG_LEVEL}

      # Enable Docker backend with default settings.
      - --providers.docker=true
      # Do not expose containers by default.
      - --providers.docker.exposedbydefault=false
      # Default Docker network used.
      - --providers.docker.network=proxy

      # --entrypoints.<name>.address for ports
      # 80 (i.e., name = webinsercure)
      - --entrypoints.webinsecure.address=:80
      # redirection to https
      - --entrypoints.webinsecure.http.redirections.entrypoint.to=websecure
      - --entrypoints.webinsecure.http.redirections.entrypoint.scheme=https
      # 443 (i.e., name = websecure)
      - --entrypoints.websecure.address=:443

      # DNS certificatesresolvers w/ Let's encrypt 
      # ACME V2 supports wildcard certificates.
      # Wildcard certificates can only be generated through a DNS-01 challenge.
      - --certificatesresolvers.leresolver.acme.dnschallenge=true
      - --certificatesresolvers.leresolver.acme.dnsChallenge.provider=godaddy
      - --certificatesresolvers.leresolver.acme.dnsChallenge.delayBeforeCheck=0
      # Email address used for registration.
      - --certificatesresolvers.leresolver.acme.email=${CERTIFICATES_EMAIL}
      # Certificates storage
      - --certificatesresolvers.leresolver.acme.storage=/letsencrypt/acme.json


      # TLS certificatesresolvers w/ Let's encrypt (for other domains such as science4fun.org)
      # See here: https://github.com/containous/traefik/issues/5632
      # Use TLS challenge for `tlsleresolver`
      - --certificatesresolvers.tlsleresolver.acme.tlschallenge=true
      # Email address used for registration.
      - --certificatesresolvers.tlsleresolver.acme.email=${CERTIFICATES_EMAIL}
      # Certificates storage (not sure if you can store all the certs in the same file)
      # Let's use a new one for the moment
      - --certificatesresolvers.tlsleresolver.acme.storage=/letsencrypt/acme_tls.json

    networks:
      # This is the network over which Traefik communicates with other containers.
      - proxy

    labels:
      ################################################
      #   Dynamic configuration with Docker Labels   #
      ################################################
      # You can tell Traefik to consider (or not) this container by setting traefik.enable to true or false.
      # We need it for the dashboard
      traefik.enable: true

      # Dashboard
      traefik.http.routers.traefik.rule: Host(`traefik.foo.org`)
      traefik.http.routers.traefik.service: api@internal
      traefik.http.routers.traefik.tls.certresolver: leresolver
      traefik.http.routers.traefik.tls.domains[0].main: foo.org
      traefik.http.routers.traefik.tls.domains[0].sans: '*.foo.org'
      traefik.http.routers.traefik.entrypoints: websecure
      # Declaring the user list
      #
      # Note=all dollar signs in the hash need to be doubled for escaping.
      # To create user:password pair, it's possible to use this command:
      # echo $(htpasswd -nb user password) and then save user and password in
      # the .env file
      traefik.http.routers.traefik.middlewares: traefik-auth
      traefik.http.middlewares.traefik-auth.basicauth.users: ${DASHBOARD_USERNAME}:${DASHBOARD_PASSWORD}

  my-test-app:
     image: containous/whoami
     networks:
      # This is the network over which Traefik communicates with other containers.
      - proxy
     labels:
      traefik.enable: true
      traefik.http.routers.my-test-app.rule: Host(`test.foo.org`)
      traefik.http.routers.my-test-app.entrypoints: websecure
      traefik.http.routers.my-test-app.tls.certresolver: leresolver
      traefik.http.routers.my-test-app.tls.domains[0].main: foo.org
      traefik.http.routers.my-test-app.tls.domains[0].sans: '*.foo.org'
      

networks:
  proxy:
    external: true
```

### If applicable, please paste the log output in DEBUG level (`--log.level=DEBUG` switch)

n/a

**Comment by Floeig** [comment]
This is how I reproduce the bug.
it's not random: Traefik gives 404 for all the routes except for one at the time
it's user/browser based and always works with curl.

You need at least 2 routes (like 2 websites).

1) on firefox when I got to the website I got 404 then press CTRL + SHIFT + R (not F5) and it change the 404 to the actual 200 and works for that website. then this website is working just fine.

2) Now when I go to the other website I get a 404 work until I press [CTRL] + [SHIFT] + [R] then it works for the new website but the previous website will now get a 404 for the user.

So it only 

1) Install all of the ingress routes definitions and rbac as you have in the doc 
2) I install a certificate and a TLS Option (not sure It's needed)
```yml
apiVersion: v1
kind: Secret
metadata:
  name: secretca1
  namespace: default
data:
    tls.ca: "__CERTIFICATE_CA__"
---

apiVersion: traefik.containo.us/v1alpha1
kind: TLSOption
metadata:
  name: mytlsoption
  namespace: default

spec:
  minVersion: VersionTLS12
  curvePreferences:
    - CurveP521
    - CurveP384
  cipherSuites:
    - TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384
    - TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
    - TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256
    - TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
    - TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305
    - TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305
  clientAuth:
    secretNames:
      - secretca1
    clientAuthType: VerifyClientCertIfGiven
  sniStrict: true
```

3) I deploy traefik
 ```yml
apiVersion: v1
kind: ServiceAccount
metadata:
  name: traefik-ingress-controller
  
---
kind: Deployment
apiVersion: apps/v1
metadata:
  name: traefik
  labels:
    app: traefik
spec:
  replicas: 2
  selector:
    matchLabels:
      app: traefik
  template:
    metadata:
      labels:
        app: traefik
    spec:
      serviceAccountName: traefik-ingress-controller
      volumes:
        - name: https-ca-certificate
          secret:
            secretName: secretca1
      restartPolicy: Always
      containers:
        - name: traefik
          image: traefik:v2.2.2
          ports:
            - name: web
              containerPort: 80
            - name: websecure
              containerPort: 443
            - name: admin
              containerPort: 8080
          args:
            - --api.insecure
            - --log.level=debug
            - --ping=true
            - --accesslog.format=json
            - --accesslog.fields.defaultmode=keep
            - --global.checknewversion
            - --global.sendanonymoususage
            - --serverstransport.insecureSkipVerify=false
            - --serverstransport.rootCAs=/cert/ca.crt
            - --entrypoints.web.Address=:80
            - --entrypoints.websecure.Address=:443
            - --providers.kubernetesCRD
          securityContext: {}
          livenessProbe:
              failureThreshold: 3
              httpGet:
                path: /ping
                port: 8080
                scheme: HTTP
              initialDelaySeconds: 10
              periodSeconds: 10
              successThreshold: 1
              timeoutSeconds: 2
          readinessProbe:
              failureThreshold: 1
              httpGet:
                path: /ping
                port: 8080
                scheme: HTTP
          volumeMounts:
            - name: https-ca-certificate
              mountPath: /cert
              readOnly: true
```
4) I create the load balancer (You probably won't locally)
 ```yml
apiVersion: v1
kind: Service
metadata: 
  name: traefik
  namespace: default
spec:
  selector:
    app: traefik
  ports:
    - protocol: TCP
      name: web
      port: 80
      targetPort: 80
    - protocol: TCP
      name: admin
      port: 8080
      targetPort: 8080
    - protocol: TCP
      name: websecure
      port: 443
      targetPort: 443
  type: LoadBalancer
  loadBalancerIP: "__MY_IP__"
```

5) I let you create 2 pod & 2 services and this is my 2 ingress routes.

```yml
apiVersion: traefik.containo.us/v1alpha1
kind: IngressRoute
metadata:
  name: traefik-web-ingressroute
  namespace: default
spec:
  entryPoints:
    - web
  routes:
    - match: Host(`abcdef.com`) # and another name for the other service
      kind: Rule
      services:
# then we need to make 2 different routes. This is an example of the services I have the issue:
        - name: api@internal
          kind: TraefikService
# or like 
        - name: myapp-service
          namespace: myapp-namespace
          port: 80
      middlewares:
        - name: traefik-websecure-redirect
          namespace: default
    
---

apiVersion: traefik.containo.us/v1alpha1
kind: IngressRoute
metadata:
  name: traefik-websecure-ingressroute
  namespace: default
spec:
  tls:
    secretName: https-certificate
    options:
      name: mytlsoption
      namespace: default
  entryPoints:
    - websecure
  routes:
    - match: Host(`abcdef.com`) # and another name for the other service
      kind: Rule
      services:
# here as well
        - name: api@internal
          kind: TraefikService
# or like 
        - name: myapp-service
          namespace: myapp-namespace
          port: 80

---

apiVersion: traefik.containo.us/v1alpha1
kind: Middleware
metadata:
  name: traefik-websecure-redirect
  namespace: default
spec:
  redirectScheme:
    scheme: https
```
that's it, you won't be able to switch between thoses 2 websites unless you do [CTRL] [SHIFT] and [R] each time

I hope It will help you a bit
Let me know if you need something else.
thanks you

_Originally posted by @Floeig in https://github.com/containous/traefik/issues/7020#issuecomment-656687017_

**Comment by floatingpurr** [comment]
I confirm that the problem affects the browsers and not curl. Thanx to @Floeig for pointing this out

**Comment by ktibi** [comment]
Hello,

Same here ! I rollback to 2.2.1 for fix.

At first I thought it came from my browser but the problem persists, even if I purge the cache or change the browser.

**Comment by SantoDE** [comment]
Actually, after further debugging and analyzation of the situation we will close that one as a duplicate of #7020. At first, it looked to be 2 different issues but apparently it seems to be the same. Sorry for the confusion. 

So closed as a duplicate of #7020
